### ISSUE TITLE
Generic immutables cannot be used as a generic type arguments inside another immutable (resulting in `<any>`)

### ISSUE THREAD

**Issue by snappieT** [immutables/immutables#848]
This is similar to #548, but not quite the same issue. That problem, which has since been fixed (thanks!) involves including a generic immutable as a field in another immutable. This issue goes one step further: when that field has the generic immutable nested as a generic type argument itself (such as a `Collection`)

For this issue, I've pushed some demonstration code, but I also summarize it below: https://github.com/snappieT/ImmutablesGenericsBug

Take one "inner" immutable, which is generic:
```java
@Value.Immutable
public interface Foo<T> {
  T getThing();
}
```

Include the generated type as a field on a second "outer" immutable, but inside a `Collection` (or any other generic wrapper, such as `Optional`):
```java
@Value.Immutable
public interface Bar {
  List<ImmutableFoo<String>> getFoos();
}
```

The generated code includes `<any>`, which then fails to compile:
```java
public final class ImmutableBar implements Bar {
  private final List<<any>> foos;
```

This occurs on version `2.7.1` (latest).

**Comment by elucash** [comment]
Thank you for reporting this bug! Apparently, fix is trivial, will be in 2.7.2

**Comment by snappieT** [comment]
Amazing, thank you so much!
